File: src/main/java/com/ada/conversation/application/dto/LlmRequest.java
package com.ada.conversation.application.dto;

import java.util.List;

public record LlmRequest(
    String model,
    List<LlmMessage> messages,
    List<LlmTool> tools,
    Double temperature,
    Integer maxTokens,
    LlmRequestMetadata metadata) {
  public LlmRequest {
    messages = List.copyOf(messages);
    tools = List.copyOf(tools);
  }

  public LlmRequest(String m, List<LlmMessage> ms, List<LlmTool> ts, LlmRequestMetadata md) {
    this(m, ms, ts, null, null, md);
  }
}

File: src/main/kotlin/com/ada/conversation/application/dto/LlmRequest.kt
package com.ada.conversation.application.dto

data class LlmRequest(
    val model: String,
    val messages: List<LlmMessage>,
    val tools: List<LlmTool> = emptyList(),
    val temperature: Double? = null,
    val maxTokens: Int? = null,
    val metadata: LlmRequestMetadata,
)
